### ISSUE TITLE
Burst of "no bencoded data to parse (84)" errors after restart of 4.0.1

### ISSUE THREAD

**Issue by JustinTArthur** [transmission/transmission#5020]
### What is the issue?

Received 30+ of these messages in the log after restarting the daemon on Linux:
```
[2023-02-23 23:07:28.968] ERR torrent-metainfo.cc:630 no bencoded data to parse (84) (torrent-metainfo.cc:630)
```

Steps:
1. Upgraded from 3.0.0 to 4.0.1
2. Restarted 4.0.1

Torrents are a mix of magnets and tracker-driven.

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.0.1 (e1c6e1be43)

**Comment by ckerr** [comment]
Hi @JustinTArthur, thanks for reporting this.

If you look in `$XDG_CONFIG_HOME/transmission/torrents`, are there any (or 30+, even :smile:) zero-byte empty `.torrent` files in there?

**Comment by ivanjx** [comment]
i got the same issue but i checked my /torrents folder and i cant find any .torrent file that has zero bytes in size

**Comment by JustinTArthur** [comment]
Right, no 0-length .torrent files in there. I did have 42 `.magnet` files in there with recent creation dates though. If those were created from the 3.x data import process, they are certainly candidates for causing this. I can't tell if the log message count lines up with these 42 as duplicate log messages past 30 are truncated.

**Comment by ckerr** [comment]
Oh, I think I see it. I'll silence the warning to avoid confusion but FWIW if my guess is right this is harmless and y'all can ignore it for now -- it is literally just the code trying to parse the files' contents as torrent metainfo before falling back and parsing them as magnet links.

**Comment by maln0ir** [comment]
Hello, I've got very similar error message after upgrading transmission-daemon from v3 do v4.0.1 on Arch Linux.

I noticed almost all of my torrents lost their stats, so even if they are completed transmission says there is no progres.  For example:

```
~ $ transmission-remote -l | grep archlinux-2023.02.01-x86_64.iso
   269    n/a       None  Done         0.0     0.0   None  Stopped      archlinux-2023.02.01-x86_64.iso
~ $ transmission-remote -t 269 --info                                                                         
NAME
  Id: 269
  Name: archlinux-2023.02.01-x86_64.iso
  Hash: 375ae3280cd80a8e9d7212e11dfaf7c45069dd35
  Magnet: magnet:?xt=urn:btih:375ae3280cd80a8e9d7212e11dfaf7c45069dd35&dn=archlinux-2023.02.01-x86_64.iso
  Labels: 

TRANSFER
  State: Stopped
  Location: /mnt/data/downloads/torrents
  Percent Done: -nan%
  ETA: 0 seconds (0 seconds)
  Download Speed: 0 kB/s
  Upload Speed: 0 kB/s
  Have: None (None verified)
  Availability: None
  Availability: -nan%
  Total size: None (None wanted)
  Uploaded: None
  Ratio: None
  Peers: connected to 0, uploading to 0, downloading from 0

HISTORY
  Date added:       Sat Feb 25 17:57:34 2023
  Date finished:    Sat Feb 25 18:13:37 2023
  Latest activity:  Sat Feb 25 18:13:37 2023
  Downloading Time: 3 minutes, 4 seconds (184 seconds)

ORIGINS
  Public torrent: Yes
  Piece Count: 0
  Piece Size: None

LIMITS & BANDWIDTH
  Download Limit: Unlimited
  Upload Limit: Unlimited
  Ratio Limit: Default
  Honors Session Limits: Yes
  Peer limit: 3
  Bandwidth Priority: Normal
```

Torrent, magnet and resume files are available and seem to be larger than 0 bytes:

```
# pwd
/var/lib/transmission/.config/transmission-daemon/
# ls -la */375ae3280cd80a8e9d7212e11dfaf7c45069dd35*
-rw------- 1 transmission transmission  5500 Feb 26 18:53 resume/375ae3280cd80a8e9d7212e11dfaf7c45069dd35.resume
-rw------- 1 transmission transmission    95 Feb 26 18:34 torrents/375ae3280cd80a8e9d7212e11dfaf7c45069dd35.magnet
-rw------- 1 transmission transmission 33154 Feb 25 18:09 torrents/375ae3280cd80a8e9d7212e11dfaf7c45069dd35.torrent
```

This ISO image is really downloaded (signature is valid).  Tried to verify or move this torrent, but it seems to take no effect, but when I started downloading again transmission crashed.  I restarted it with `--log-debug` option and this is what `journalctl` output looks like:

```
Feb 26 19:26:22 Nidhoggr transmission-daemon[73960]: [2023-02-26 19:26:22.830] dbg archlinux-2023.02.01-x86_64.iso metadata is 33078 bytes in 3 pieces (/build/transmission/src/transmission-4.0.1/libtransmission/torrent-magnet.cc:86)
<cut similar messages>
Feb 26 19:26:29 Nidhoggr transmission-daemon[73960]: [2023-02-26 19:26:29.830] dbg archlinux-2023.02.01-x86_64.iso saving metainfo piece 2... 1 remain (/build/transmission/src/transmission-4.0.1/libtransmission/torrent-magnet.cc:385)
Feb 26 19:26:30 Nidhoggr transmission-daemon[73960]: /usr/include/c++/12.2.1/bits/stl_algo.h:3623: constexpr const _Tp& std::clamp(const _Tp&, const _Tp&, const _Tp&) [with _Tp = short unsigned int]: Assertion '!(__hi < __lo)' failed.
Feb 26 19:26:30 Nidhoggr systemd[1]: transmission.service: Main process exited, code=dumped, status=6/ABRT
Feb 26 19:26:30 Nidhoggr systemd[1]: transmission.service: Failed with result 'core-dump'.
Feb 26 19:26:30 Nidhoggr systemd[1]: transmission.service: Consumed 1.320s CPU time.
```

**Comment by ckerr** [comment]
@maln0ir Thank you for reporting that! That's related to 5020 but is already covered by #5007 and is fixed in #5025

**Comment by alvindarkness** [comment]
In case it helps, I just want to add that this behaviour (the burst "no bencoded data to parse" logged lines) was not present in 4.0.0, and only started in 4.0.1.  I can readily switch between the two and will only see it occur on 4.0.1.

I only add this since no other comments have mentioned it yet (they've all gone from 3.x to 4.0.1).

**Comment by alvindarkness** [comment]
Just a quick update, compiled and started up 4.0.2 and I no longer experience the burst of "no bencoded data to parse" log lines.  Will report back if it returns after a restart in the future.  Might be good to see if 4.0.2 has fixed the issue for others also.

**Comment by JustinTArthur** [comment]
@ckerr looks to have fixed this in #5025 by separating the .magnet and .torrent handling. With this, .magnet files are no longer sent up the benc toolchain. Someone with write permission can link this issue to that PR.

I suspect there's no reason to silence the warning as with the underlying bug fixed, the warning is useful again. You can re-open if I'm wrong, @ckerr.
